[stable/datadog] Support 1.6 tolerations

vreon opened the pull request:
This allows the Datadog DaemonSet to run on tainted nodes, whether
that's a master provisioned by kubeadm (see #1100):

```yaml
tolerations:
  - effect: NoSchedule
    key: node-role.kubernetes.io/master
```

or any "dedicated" node:

```yaml
tolerations:
  - effect: NoSchedule
    operator: Exists
```

Commit:
[stable/datadog] Support 1.6 tolerations

This allows the Datadog DaemonSet to run on tainted nodes, whether
that's a master provisioned by kubeadm (see #1100):

    tolerations:
      - effect: NoSchedule
        key: node-role.kubernetes.io/master

or any "dedicated" node:

    tolerations:
      - effect: NoSchedule
        operator: Exists

lachie83 commented:
@k8s-bot ok to test

lachie83 commented on stable/datadog/README.md:
```diff
@@ -53,6 +53,7 @@ The following tables lists the configurable parameters of the Datadog chart and
 | `resources.limits.cpu`      | CPU resource limits                | 512Mi                                     |
 | `resources.requests.memory` | Memory resource requests           | 100m                                      |
 | `resources.limits.memory`   | Memory resource limits             | 256m                                      |
+| `daemonset.tolerations`     | List of node taints to tolerate    | `nil`                                     |
```
Can you call out that this is a 1.6 or greater feature?

vreon replied:
Can do! Do you think it would be worth it to add a `daemonset.podAnnotations` value to accommodate 1.5 clusters?

lachie83 replied:
Yeah I think that's a good idea. Just add some examples in values.yaml

Commit:
[stable/datadog] Document 1.6 requirement for `tolerations`

Commit:
[stable/datadog] Add `daemonset.podAnnotations` value

This allows Kubernetes 1.5 users to use the alpha tolerations annotation
to schedule DataDog Pods on tainted nodes.

Commit:
[stable/datadog] Docs tweak: greater than 1.6

Commit:
[stable/datadog] Docs tweak: >= 1.6
